packages/app/src/runs/RunsContainer.vue

```diff
@@ -71,11 +95,29 @@ const props = defineProps<{
   gql: RunsContainerFragment
 }>()
 
+const isCloudProjectTypename = props.gql.currentProject?.cloudProject?.__typename === 'CloudProject'
+
+const isShowingOnlineNotification = ref(false)
+const isOnlineRef = ref(true)
 const showConnectSuccessAlert = ref(false)
 
+watchEffect(() => {
+  // We want to keep track of the previous state to refetch the query
+  // when the internet connection is back
+  if (!online.value && isOnlineRef.value) {
+    isOnlineRef.value = false
+  }
+
+  if (online.value && !isOnlineRef.value) {
+    isOnlineRef.value = true
+    emit('reexecuteRunsQuery')
+  }
+})
+
+const isCloudProjectReturned = computed(() => isCloudProjectTypename)
 const currentProject = computed(() => props.gql.currentProject)
 const cloudViewer = computed(() => props.gql.cloudViewer)
-
+const isOnline = computed(() => online.value)
```

Comment:
This doesn't do anything - `isOnline` is already a `ref` so `watchEffect` will trigger just fine, or at least, it should.